app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/ui/MainActivity.java

```diff
@@ -87,7 +89,27 @@ protected void onCreate(Bundle savedInstanceState) {
         FragmentManager fragmentManager = getSupportFragmentManager();
         mNavigationDrawerFragment = (NavigationDrawerFragment)
                 fragmentManager.findFragmentById(R.id.navigation_drawer);
-        checkSDCardData();
+
+        // Update pattern cache
+        // note, when we move to RxJava 2 this Observable can be replaced with a Maybe
+        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(this);
+        final long lastUpdated = sp.getLong(LINES_LAST_UPDATED, -1);
+        Observable.just(lastUpdated)
+                .delay(10,TimeUnit.SECONDS)
+                .map(t -> patApiService
+                        .getPatternDataManager()
+                        .updatePatternCache(System.currentTimeMillis(), t))
+                .filter(t -> t != lastUpdated)
+                .subscribeOn(Schedulers.computation())
```

Comment:
Is this better done on the IO scheduler? I know we're clearing cache on 1 thread but it may be beneficial to put it on the IO thread so we will only have 1 background thread looking at the disk.